conda_libmamba_solver/solver.py

```diff
@@ -527,73 +552,98 @@ def _problems_to_specs_parser(problems: str) -> Mapping[str, MatchSpec]:
 
         if (previous and (previous_set == current_set)) or len(diff) >= 10:
-            # We have same or more (up to 10) conflicts now! Abort to avoid recursion.
-            self._raise_for_problems(problems)
-
-        return conflicts
-
-    def _raise_for_problems(self, problems: Optional[str] = None):
-        if self.solver is None:
-            raise RuntimeError("Solver is not initialized. Call `._setup_solver()` first.")
-
-        if problems is None:
-            problems = self.solver.problems_to_str()
-
-        # TODO: Figure out a way to have ._problems_to_specs_parser
-        # return the most adequate exception type instead of reparsing here
-        missing_from_channel = []
-        for line in problems.splitlines():
-            line = line.strip()
-            if line.startswith("- nothing provides requested"):
-                packages = line.split()[4:]
-                missing_from_channel += packages
-        if missing_from_channel:
-            exc = PackagesNotFoundError(missing_from_channel, list(self.channels))
-        else:
-            exc = LibMambaUnsatisfiableError(problems)
-
-        # do not allow conda.cli.install to try more things
-        exc.allow_retry = False
-        raise exc
+            # We have same or more (up to 10) unsatisfiable now! Abort to avoid recursion
+            message = self._prepare_problems_message()
+            exc = LibMambaUnsatisfiableError(message)
+            # do not allow conda.cli.install to try more things
+            exc.allow_retry = False
+            raise exc
+        return unsatisfiable
+
+    def _prepare_problems_message(self):
+        if hasattr(self.solver, "explain_problems"):
+            try:
+                raw_problems = self.solver.explain_problems()
+                key = None
+                sections = {}
+                for line in raw_problems.splitlines():
+                    if line.strip().startswith("===="):
+                        title = line.strip("=").strip()
+                        if "(old)" in title:
+                            key = "old"
+                        elif "(new)" in title:
+                            key = "new"
+                        else:
+                            key = "disclaimer"
+                        sections[key] = []
+                        continue
+                    if key is not None:
+                        sections[key].append(line)
+                return "\n".join(
+                    [
+                        *sections["old"],
+                        "",
+                        "***************************************************",
+                        " conda has set experimental_sat_error_message=true",
+                        " You will now see an explanation of the conflicts.",
+                        " You can provide feedback to the libmamba team at:",
+                        "  https://github.com/mamba-org/mamba/issues/2078",
```

Comment:
```suggestion
                        " https://github.com/mamba-org/mamba/issues/2078",
```

Is there prior art for how these block notifications are rendered in conda to make it less block-y?

Reply:
I'd say prior art is even block-ier: 

https://github.com/conda/conda-libmamba-solver/blob/dc6b266526a1ab6e07fd01144b6bf420c6f00fd1/conda_libmamba_solver/solver.py#L841-L851

Reply:
The leading space was there to make the URL centered and distinguishable from the paragraph text, but happy to change it.

Reply:
Nah, it's fine